src/include/sof/wait.h

```diff
@@ -145,6 +146,25 @@ static inline int wait_for_completion_timeout(completion_t *comp)
 	}
 }
 
+static inline int wait_for_completion_idelay(int (*condition)(void *data), void *data, int clk)
+{
+	int try = PLATFORM_DEFAULT_TRIES;
+
+	while (!condition(data)) {
+		idelay(clk);
```

Comment:
this must be divided by tries. otherwise we will wait for multiples of the timeout and not the timeout.

Reply:
yes, I will refine my patch